packages/jellyfish-transaction/src/script/dftx/dftx_icxorderbook.ts

```diff
@@ -0,0 +1,50 @@
+import { BufferComposer, ComposableBuffer } from '../../buffer/buffer_composer'
+import BigNumber from 'bignumber.js'
+import { Script } from '../../tx'
+import { CScript } from '../../tx_composer'
+
+// Disabling no-return-assign makes the code cleaner with the setter and getter */
+/* eslint-disable no-return-assign */
+
+export enum ICXOrderType {
+  /** type for DFI/BTC orders */
+  INTERNAL = 1,
+  /** type for BTC/DFI orders */
+  EXTERNAL = 2
+}
+
+/**
+ * ICX CreateOrder DeFi Transaction
+ */
+
+export interface ICXCreateOrder {
+  orderType: number // -------------| 1 byte unsigned
+  tokenId: number // ---------------| VarUInt{1-9 bytes}
+  ownerAddress: Script // ----------| n = VarUInt{1-9 bytes}, + n bytes
+  receivePubkey?: string // --------| 1 byte for len + len bytes
+  amountFrom: BigNumber // ---------| 8 bytes unsigned
```

Comment:
Shall we rename this to `amount`? since it is not INTERNAL/EXTERNAL specific

Reply:
I'd rather we keep amountFrom as in ICXOrder interface and CICXOrder class in `icxorder.h`